full-service/src/json_rpc/v2/e2e_tests/transaction/build_submit/build_then_submit.rs

```diff
@@ -717,10 +717,18 @@ mod e2e_transaction {
 
     #[test_with_logger]
     fn test_reuse_input_txo_then_submit_transaction(logger: Logger) {
+        // This test mimics a situation where multiple services are using the same
```

Comment:
I decided to modify the test that was added as part of #738 since I think that the new code logic better captures the intended behavior and it seems like it's very similar if not the same problem that #738 was trying to solve.

I'm a big proponent of tests, but I'm also a big proponent of, just enough tests.